x/poolmanager/client/query_proto_wrap.go

```diff
@@ -142,18 +142,16 @@ func (q Querier) EstimateSwapExactAmountOutWithPrimitiveTypes(ctx sdk.Context, r
 
 func (q Querier) EstimateSinglePoolSwapExactAmountOut(ctx sdk.Context, req queryproto.EstimateSinglePoolSwapExactAmountOutRequest) (*queryproto.EstimateSwapExactAmountOutResponse, error) {
 	routeReq := &queryproto.EstimateSwapExactAmountOutRequest{
-		PoolId:   req.PoolId,
 		TokenOut: req.TokenOut,
-		Routes:   types.SwapAmountOutRoutes{{PoolId: req.PoolId, TokenInDenom: req.TokenInDenom}},
+		Routes:   types.SwapAmountOutRoutes{{TokenInDenom: req.TokenInDenom}},
```

Comment:
This change doesn't follow for me.. I can see why you wouldn't need the PoolId field, but the PoolId within the Routes feels necessary, unless I am missing something?

Reply:
Its a single pool is why

Reply:
I actually think we need this field on the second thought, we've erased the pool id field since there was pool id in request struct itself, this pool id is from the routes itself. Good catch, recommitted change

Reply:
Ah I do see the point, we're deprecating pool id field due to it being in the routes for normal swaps, but for single swaps, this isn't the case, we're getting pool id as input and constructing it as routes within query proto wrap so we should indeed be having the pool id field **only** for single swap estimate

Reply:
Changes introduced in https://github.com/osmosis-labs/osmosis/pull/6530/commits/600209d1fc265fa6a754279089caf39017926f72

Reply:
Right but we were removing BOTH pool ids, you need at least one of them to determine what pool we are in, otherwise we would just have a denom. I think you change addresses this

Reply:
yeah exactly!